lib/web_ui/lib/src/engine/pointer_binding/event_position_helper.dart

```diff
@@ -0,0 +1,113 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:ui/ui.dart' as ui show Offset;
+
+import '../dom.dart';
+import '../semantics.dart' show EngineSemanticsOwner;
+
+/// Returns an [ui.Offset] of the position of [event], relative to the position of [actualTarget].
+///
+/// The offset is *not* multiplied by DPR or anything else, it's the closest
+/// to what the DOM would return if we had currentTarget readily available.
+///
+/// This needs an `actualTarget`, because the `event.currentTarget` (which is what
+/// this would really need to use) gets lost when the `event` comes from a "coalesced"
+/// event.
+///
+/// It also takes into account semantics being enabled to fix the case where
+/// offsetX, offsetY == 0 (TalkBack events).
+//
+// TODO(dit): Make this understand 3D transforms in the platform view case, https://github.com/flutter/flutter/issues/117091
```

Comment:
Should we also teach it to understand 3D transforms in the semantics tree? (not that people use this often, but for completeness)

Reply:
Ah yes, I should have removed the "in the platform view case" bit of text when I moved this from the other file. I think we need to solve the 3D case for all the use cases where we cannot use directly offsetX/offsetY (that comes pre-computed by the browser)

Reply:
I've moved the TODOs to a better spot.